### PR TITLE
Feature/delete stories

### PR DESCRIPTION
**Pull request by belengg27**

### CODE REVIEW

File: app/src/main/java/ch/proximeety/proximeety/presentation/views/stories/StoriesView.kt [hunk 1]
```diff
@@ -94,30 +93,39 @@ fun StoriesView(
...
-                            .background(Color.Gray)
-                            .clickable { viewModel.onEvent(StoriesEvent.OnClickOnUserPicture) }
-                    )
-                    Spacer(modifier = Modifier.width(MaterialTheme.spacing.small))
-                    Column {
-                        Text(
-                            text = user.value?.displayName.toString(),
-                            style = MaterialTheme.typography.h4
+                    Row() {
+                        Spacer(modifier = Modifier.width(MaterialTheme.spacing.small))
```
Comment (yanisdb) [inline app/src/main/java/ch/proximeety/proximeety/presentation/views/stories/StoriesView.kt:100]: Don't put a spacer. If you want to leave more spacer using `Modifier.padding(start = ...)` on the `Row`.

File: app/src/main/java/ch/proximeety/proximeety/presentation/views/stories/StoriesView.kt [hunk 2]
```diff
@@ -94,30 +93,39 @@ fun StoriesView(
                     modifier = Modifier.padding(
                         vertical = MaterialTheme.spacing.extraSmall,
                         horizontal = MaterialTheme.spacing.small
-                    )
+                    ).fillMaxWidth(),
+                    horizontalArrangement = Arrangement.SpaceBetween
```
Comment (yanisdb) [inline app/src/main/java/ch/proximeety/proximeety/presentation/views/stories/StoriesView.kt:97]: This is where you should add the `verticalAlignment = Alignment.CenterVertically`.

File: app/src/main/java/ch/proximeety/proximeety/presentation/views/stories/components/ButtonExtended.kt [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    val menuExpanded = remember { mutableStateOf(false) }
+    val showDialog = remember { mutableStateOf(false) }
+
+    val options = arrayOf("Delete Story", "More")
+
+    IconButton( onClick = { menuExpanded.value = true }) {
+        Icon(imageVector = Icons.Rounded.MoreHoriz, contentDescription = "More")
+    }
+
+    Column() {
```
Comment (yanisdb) [inline app/src/main/java/ch/proximeety/proximeety/presentation/views/stories/components/ButtonExtended.kt:30]: You don't need this `Column`. There is only elements inside and no `Modifier`.

File: app/src/main/java/ch/proximeety/proximeety/presentation/views/stories/components/ButtonExtended.kt [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+import ch.proximeety.proximeety.presentation.views.stories.StoriesViewModel
+
+@SuppressLint("StateFlowValueCalledInComposition")
+@Composable
+fun ButtonExtended(viewModel: StoriesViewModel, onDelete: () -> Unit){
+    val menuExpanded = remember { mutableStateOf(false) }
+    val showDialog = remember { mutableStateOf(false) }
+
+    val options = arrayOf("Delete Story", "More")
+
```
Comment (yanisdb) [inline app/src/main/java/ch/proximeety/proximeety/presentation/views/stories/components/ButtonExtended.kt:25]: Add a `Column(horizontalAlignment = Alignment.End)` and close it your `DropDownMenu`.

**Review by yanisdb — CHANGES_REQUESTED** [review]
Hello,

This is a very good pull request. 

### UI
One of the few things I could find that you could improve is the UI :The button is not exactly at the right place.

Screenshot_20220428-090906_Proximeety (1)

This is caused by two things. 
1. It is too much to the left. This comes from your `ButtonExtended` composable. You don't wrap the `IconButton` and the `DropdownMenu` in anything, so by default, Jetpack Compose will try to fit them both. To fix that, you should wrap your `IconButton` and your `DropdownMenu` in a `Column` and set the `horizontalAlignment` to `Alignment.End`.

2.  The button is too far down. To fix that, we could bring it up, but this would be bad practice. The default padding is there for a reason. Another way we could fix it is by bringing the name down. We can do that easily by setting the `vertialAlignment` property of the `Row` containing both elements to `Alignment.CenterVerticaly`.

I quickly tried it, but I'm going to let you do it. I also aded some line comments so that you know where to make the changes. The final result should look something like that:
 
Screenshot_20220428-092015_Proximeety

### General remarks

* Be sure to format your code before committing. You can make this happens automatically by going in `Settings > Version Control > Commit Select` and right under `Before Commit` you should check the `Reformat code` checkbox. You can also check `Rearrange code`, `Optimize imports` and `Cleanup`.
* The simple `SimpleAlertDialog` is probably going to be used extensively and in other parts of the code. So maybe in shouldn't be in the `components` package of a view. To make it a general component, we could put it in `ch.proximeety.proximeety.presentation.components`. If you make it a general component, the title and the question should be parameters.

**Review by yanisdb — APPROVED** [review]

**Comment by yanisdb** [comment]
As these are not issue with the quality of the code but rather with the quality of the user interfance, these changes can be addressed in a later sprint. If @belengg27 has time, she will fix it next week.
Thus, I approve the pull request.

**Review by Hamzezi — APPROVED** [review]
Great Pull request again!

**Pull request merged**